tokens/component/color/dark/avatar.json5

```diff
@@ -0,0 +1,25 @@
+{
```

Comment:
The fragmentation and nesting makes token source feel overwhelming and complex. Specifically:
- having 1 file per component
- requiring more levels of folder nesting

I'd look for opportunities to reduce the number of files and levels of nesting.

### Example

#### Before

```
tokens/
├─ component/
│  ├─ color/
│  │  ├─ dark/
│  │  │  ├─ components-dark.json5
│  │  │  ├─ avatar.json5
│  │  │  ├─ counter.json5
│  │  │  ├─ overrides/
│  │  │  │  ├─ dark.high-contrast.json5
│  │  │  ├─ topicTag.json5
├─ reference/
│  ├─ color/
│  │  ├─ dark/
│  │  │  ├─ dark.dimmed.json5
│  │  │  ├─ dark.high-contrast.json5
│  │  │  ├─ dark.json5
│  │  │  ├─ dark.protanopia-deuteranopia.json5
│  │  │  ├─ dark.protanopia-tritanopia.json5
│  ├─ size/
│  │  ├─ size.json5
│  ├─ typography/
│  │  ├─ typography.json5
├─ system/
│  ├─ color/
│  │  ├─ dark/
│  │  │  ├─ ansi-dark.json5
│  │  │  ├─ checks-dark.json5
│  │  │  ├─ diff-dark.json5
│  │  │  ├─ marketting-dark.json5
│  │  │  ├─ misc-dark.json5
│  ├─ size/
│  │  ├─ border.json5
│  │  ├─ breakpoint.json5
│  │  ├─ size-coarse.json5
│  │  ├─ size-fine.json5
│  │  ├─ size.json5
│  │  ├─ viewport.json5
│  ├─ typography/
│  │  ├─ typography.json5
```

#### After
```
tokens/
├─ color/
│  ├─ components/
│  │  ├─ dark.json5
│  │  ├─ dark.overrides.json5
│  ├─ reference/
│  │  ├─ dark.dimmed.json5
│  │  ├─ dark.high-contrast.json5
│  │  ├─ dark.json5
│  │  ├─ dark.protanopia-deuteranopia.json5
│  │  ├─ dark.protanopia-tritanopia.json5
│  ├─ system/
│  │  ├─ dark.ansi.json5
│  │  ├─ dark.checks.json5
│  │  ├─ dark.diff.json5
│  │  ├─ dark.marketting.json5
│  │  ├─ dark.misc.json5
├─ size/
│  ├─ components.json5
│  ├─ reference/
│  │  ├─ border.json5
│  │  ├─ breakpoint.json5
│  │  ├─ dimensions.json5
│  │  ├─ dimensions-coarse.json5
│  │  ├─ dimensions-fine.json5
│  │  ├─ viewport.json5
├─ typography/
│  ├─ reference.json5
│  ├─ system.json5
```

Reply:
Hey @mperrotti,
thank you for the detailed reply. 🙏 

I agree with some ideas and am unsure about others, so I hope this comment will be understandable. 😉 

>  Specifically:
>- having 1 file per component
I find this to be helpful as it makes it much easier to see which components have custom components. However I agree that this adds a lot of files that can be overwhelming.

My main concern is that if we have components that have overrides for color modes this could be a horror scenario of one property files. 👻 

> - requiring more levels of folder nesting
I think it is good to visualize it, but in the end I do agree that having a reduced folder set is nicer.

```
tokens/
├─ color/
│  ├─ components/
│  │  ├─ dark.json5
│  │  ├─ dark.overrides.json5
│  ├─ reference/
│  │  ├─ dark.dimmed.json5
│  │  ├─ dark.high-contrast.json5
│  │  ├─ dark.json5
│  │  ├─ dark.protanopia-deuteranopia.json5
│  │  ├─ dark.protanopia-tritanopia.json5
│  ├─ system/
│  │  ├─ dark.ansi.json5
│  │  ├─ dark.checks.json5
│  │  ├─ dark.diff.json5
│  │  ├─ dark.marketting.json5
│  │  ├─ dark.misc.json5
├─ size/
│  ├─ components.json5
│  ├─ reference/
│  │  ├─ border.json5
│  │  ├─ breakpoint.json5
│  │  ├─ dimensions.json5
│  │  ├─ dimensions-coarse.json5
│  │  ├─ dimensions-fine.json5
│  │  ├─ viewport.json5
├─ typography/
│  ├─ reference.json5
│  ├─ system.json5
```

I do like this approach and was thinking about something like this as well. Getting rid of the `dark` and `light` folders made a lot of sense once I added the `dark.` prefix. 

The thing that I discussed with I think @simurai before and am not sure about is using a `reference/` folder in one places and a `reference.json5` file in others. I like it, but I am afraid that this may be confusing for some people.

Do we do something like this in other places?

An alternative could be something like this:

```
├─ size/
│  ├─ components/
│  │ ├─ components.json5
│  ├─ reference/
```